### PR TITLE
Newest: sync with BE #67

### PR DESCRIPTION
**Pull request by Gtes**
Newest: sync with BE #67

## GitHub Board

**Issue link**

[#67  ]( Newest: sync with BE)

## Summary of issue

- [x] Data to the newest section is fetched from the BE 
- [x] Use axios to perform the request
- [x] use thunk action to perform the request 
- [x] use redux to store the data
- [x] use redux to get the data to the container component

## Summary of change
- Added spinner for loading status. [screenshot](https://i.imgur.com/yEfYVJH.png)
- Added placeholder if the request failed. [screenshot](https://i.imgur.com/IKkyqDr.png)
- Hide Load More button while the request is loading.
- Added placeholder if the last page loaded. [screenshot](https://i.imgur.com/ws9gVRI.png)

### CODE REVIEW

File: src/lib/components/PostPreview/PostPreviewCard.tsx [hunk 1]
```diff
@@ -67,19 +67,19 @@ const PostPreviewCard: React.FC<IPostPreviewCardProps> = (props) => {
               variant="subtitle2"
               component="h3"
               style={{
-                backgroundColor: DIRECTION_PROPERTIES[data.direction].color,
+                // backgroundColor: DIRECTION_PROPERTIES[data.direction].color,
```
Comment (Gtes) [inline src/lib/components/PostPreview/PostPreviewCard.tsx:70]: Will be fixed in #95

File: src/modules/main/components/NewestContainer.tsx [hunk 2]
```diff
@@ -1,53 +1,93 @@
...
+    return state.main.newest;
+  });
 
   useEffect(() => {
     setNewest();
   }, []);
 
+  const renderLoading = () => {
+    return loading === 'pending' ? <CircularProgress /> : null;
+  };
```
Comment (SergeOlabin) [inline src/modules/main/components/NewestContainer.tsx:36]:
```suggestion
  const renderLoading = () =>loading === 'pending' ? <CircularProgress /> : null;
```
1-liners are better here :)

File: src/modules/main/components/NewestContainer.tsx [hunk 3]
```diff
@@ -1,53 +1,93 @@
...
+
+  const renderError = (
+    errorMsg = 'Не вдалося завантажити більше матеріалів',
+  ) => {
+    return loading === 'failed' ? <span>{errorMsg}</span> : null;
+  };
```
Comment (SergeOlabin) [inline src/modules/main/components/NewestContainer.tsx:42]:
```suggestion
  const renderError = (
    errorMsg = 'Не вдалося завантажити більше матеріалів',
  ) => loading === 'failed' ? <span>{errorMsg}</span> : null;
```

File: src/modules/main/store/mainSlice.ts [hunk 4]
```diff
@@ -9,22 +9,29 @@ import {
...
 import { getPosts } from '../../../lib/utilities/API/api';
 import MOCK_EXPERTS from '../mockDataExperts';
-import MOCK_NEWEST from '../components/constants/newestPosts-mock';
 import type { AppThunkType } from '../../../store/store';
 import { LOAD_POSTS_LIMIT } from '../components/constants/newestPostsPagination-config';
+import type { RootStateType } from '../../../store/rootReducer';
+
+const POST_PREVIEW_LENGTH = 150;
 
 interface IMeta {
```
Comment (SergeOlabin) [inline src/modules/main/store/mainSlice.ts:18]: INewestMeta

File: src/modules/main/store/mainSlice.ts [hunk 5]
```diff
@@ -35,13 +42,56 @@ const initialState: IMainState = {
...
+        'lastName',
+        'mainInstitution',
+      ]) as IExpert;
+
+      const preview = _.truncate(post.content, {
+        length: POST_PREVIEW_LENGTH,
+      });
+
+      return {
+        author: { ...postAuthor, workPlace: postAuthor.mainInstitution?.name },
```
Comment (SergeOlabin) [inline src/modules/main/store/mainSlice.ts:81]: can you `pick` workPlace on line:74?

**Review by SergeOlabin — COMMENTED** [review]
let's talk on sync

File: src/modules/main/store/mainSlice.ts (hunk 5 above)
Comment (Gtes) [reply src/modules/main/store/mainSlice.ts]: fixed

File: src/modules/main/store/mainSlice.ts (hunk 4 above)
Comment (Gtes) [reply src/modules/main/store/mainSlice.ts]: fixed

File: src/modules/main/components/NewestContainer.tsx (hunk 3 above)
Comment (Gtes) [reply src/modules/main/components/NewestContainer.tsx]: fixed

File: src/modules/main/components/NewestContainer.tsx (hunk 2 above)
Comment (Gtes) [reply src/modules/main/components/NewestContainer.tsx]: fixed

File: src/lib/types.ts [hunk 6]
```diff
@@ -39,6 +46,7 @@ export interface IPost {
   modifiedAt?: string;
   postType: PostTypeEnum;
   preview?: string;
+  id?: number;
```
Comment (SergeOlabin) [inline src/lib/types.ts:49]: it should be required.
do we have any cases when it's optional?

**Review by SergeOlabin — APPROVED** [review]

File: src/lib/types.ts (hunk 6 above)
Comment (Gtes) [reply src/lib/types.ts:49]: Yep, in `fetchImportantPosts` thunk, we don't use the `id` field.

[fetchImportantPosts](https://github.com/ita-social-projects/dokazovi-fe/blob/4530fd690bfe3d48a9269323171f14b4768562ed/src/modules/main/store/mainSlice.ts#L135-L159)

**Pull request merged**